tree/treeplayer/src/TTreeProcessorMT.cxx

```diff
@@ -328,26 +328,33 @@ Internal::FriendInfo TTreeProcessorMT::GetFriendInfo(TTree &tree)
 
    for (auto fr : *friends) {
       const auto frTree = static_cast<TFriendElement *>(fr)->GetTree();
+      const bool isChain = frTree->IsA() == TChain::Class();
 
-      // Check if friend tree has an alias
-      const auto realName = frTree->GetName();
-      const auto alias = tree.GetFriendAlias(frTree);
-      if (alias) {
-         friendNames.emplace_back(std::make_pair(realName, std::string(alias)));
-      } else {
-         friendNames.emplace_back(std::make_pair(realName, ""));
-      }
-
-      // Store the file names of the friend tree
       friendFileNames.emplace_back();
       auto &fileNames = friendFileNames.back();
-      const bool isChain = tree.IsA() == TChain::Class();
+
       if (isChain) {
-         const auto frChain = static_cast<TChain *>(frTree);
-         for (auto f : *(frChain->GetListOfFiles())) {
+         const auto chainFiles = static_cast<TChain*>(frTree)->GetListOfFiles();
+         // TODO: This does not yet take into account different names for each tree in the chain.
+         // This needs a refactoring of the FriendInfo object. In the chainFiles, each object
+         // has the attribute fName with the name of the tree and the attribute fTitle with the
+         // name of the respective file.
+         const auto realName = chainFiles->First()->GetName();
```

Comment:
Can you please add a comment explaining that for the case in which the name of the TTree is contained as part of the filename, TFile::GetName happens to return the TTree name, and TChain::GetName is empty/useless? It's really obscure :sweat_smile:

Reply:
It is :D I put a comment :)

Reply:
It looks to me like the comment is over logic that is the same as before: `for (auto f: chain->GetListOfFiles()) fileNames.emplace_back(f->GetTitle())`, or equivalent.

I think the obscure part is that `realName` is `chainFiles->First()->GetName()` instead of `frTree->GetName()`.

Related: does `chain->GetListOfFiles()->First()->GetName()` (note that `GetName()` is called on a `TFile`) return the name of the `TTree` also in the case in which the tree name is _not_ part of the filename?

Reply:
Yep, it does. I put such a case in the tests!